tasks/retrieval/msrvtt/retrieval_msrvtt_dataset.py

```diff
@@ -0,0 +1,130 @@
+from typing import Optional, Any, Dict, List, Tuple
+import os
+import json
+import pandas as pd
+import numpy as np
+from torch.utils.data import Dataset
+from PIL import Image
+from flagevalmm.registry import DATASETS
+from flagevalmm.common.const import FLAGEVALMM_DATASETS_CACHE_DIR
+from flagevalmm.dataset.utils import get_data_root
+from rawframe_util import RawFrameExtractor
+
+@DATASETS.register_module()
+class RetrievalMSRVTTDataset(Dataset):
+    def __init__(
+        self,
+        *,
+        name: str,
+        data_root: Optional[str] = None,
+        anno_file: Optional[str] = None,
+        cache_dir: str = FLAGEVALMM_DATASETS_CACHE_DIR,
+        config: Optional[dict] = None,
+        base_dir: Optional[str] = None,
+        max_words: int = 30,
+        feature_framerate: float = 1.0,
+        max_frames: int = 100,
+        image_resolution: int = 224,
+        tokenizer=None,  # Add tokenizer parameter
+        **kwargs,
+    ) -> None:
+        self.data_root = get_data_root(
+            data_root=data_root, config=config, cache_dir=cache_dir, base_dir=base_dir
+        )
+
+        self.name = name
+        # Load the MSRVTT dataset CSV file
+        self.csv_path = os.path.join(self.data_root, anno_file if anno_file else "MSRVTT_JSFUSION_test.csv")
+        self.data = pd.read_csv(self.csv_path)
+
+        # Initialize the video frame extractor
+        self.frameExtractor = RawFrameExtractor(framerate=feature_framerate, size=image_resolution)
+
+        # Initialize text and video-related parameters
+        self.max_words = max_words
+        self.max_frames = max_frames
+        self.feature_framerate = feature_framerate
+        self.image_resolution = image_resolution
+        self.tokenizer = tokenizer  # Set tokenizer
+
+        # Load video and caption data
+        self.videos = self.data['video_id'].values
+        self.captions = self.data['sentence'].values
+
+    def __len__(self):
+        return len(self.videos)
+
+    def __getitem__(self, index: int) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        video_id = self.videos[index]
+        caption = self.captions[index]
+
+        # Get video frames
+        video_frames, video_mask = self._get_rawvideo(video_id)

```

Comment:
There is no need to read video in the dataset, it just provide video path and caption.